### PR TITLE
fix: Update action bar row layout in UserManagement page

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Ticket
[ET-299](https://hpe-aiatscale.atlassian.net/browse/ET-299)

## Description

The user management page had a broken layout for the "add user" button on mobile view. This PR intends to fix that.

## Test Plan
- Log in as an admin
- go to the "Admin" management page from the dropdown menu (that should take you to the user management page)
- enable mobile view (from the terminal console, there is a "desktop" button on the top-left corner, that toggles the view mode)
- the UI should have the "add user" button showing below the filters

Screenshot 2024-08-27 at 16 31 49

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-299]: https://hpe-aiatscale.atlassian.net/browse/ET-299?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keita-determined — COMMENTED** [review]
can we fix the horizontal scroll issue? 

https://github.com/user-attachments/assets/a97855b4-e334-4944-a71d-740cd4432260

--- 

also can we confirm with a designer and a product manage if its a good approach to fix the ticket issue r? not sure if we wanna show `add user` button in new line.

**Comment by keita-determined** [comment]
> I believe that this is an animation from the browser, I've tested locally and got that animation too, but, no scroll is being kept...

i dont think so. It doesn't bounce, i can just scroll horizontally.

File: webui/react/src/pages/Admin/UserManagement.tsx [hunk 1]
```diff
@@ -483,32 +485,35 @@ const UserManagement: React.FC<Props> = ({ onUserCreate }: Props) => {
...
                   onChange={handleNameSearchApply}
                 />
                 <Select
                   data-testid="roleSelect"
                   options={roleOptions}
+                  placeholder="Role"
                   searchable={false}
                   value={roleFilter}
-                  width={120}
+                  width={isMobile ? 'calc(100vw - padding)' : 120}
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/UserManagement.tsx:507]: where does `padding` come from?

**Review by keita-determined — COMMENTED** [review]
the issue is not fixed. could you please test it before requesting the review?

**Review by alexjdetermined — APPROVED** [review]
LGTM

File: webui/react/src/pages/Admin/UserManagement.tsx (hunk 1 above)
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.tsx:507]: Please ensure this code is actually running in mobile -- i'm not seeing this style (or the other calc style) applied when in mobile.

**Review by keita-determined — COMMENTED** [review]
nit: button layout looks inconsistent with different viewports.

https://github.com/user-attachments/assets/1675b508-2742-41ce-9e36-67a1608bf1bc

**Review by keita-determined — COMMENTED** [review]
image

the issue is still present

**Comment by keita-determined** [comment]
that needs to be desktop mode instead of mobile. i think It should need the same treatment as mobile.

**Comment by keita-determined** [comment]
> > that needs to be desktop mode instead of mobile. i think It should need the same treatment as mobile.
> 
> so, this is the smallest desktop breakpoint
> 
> image

on desktop 800 x 950

on mobile 
image

**Comment by keita-determined** [comment]
can we move `add user` button to the bottom (the same row as status select) in tablet view to make all views consistent?

**Review by keita-determined — COMMENTED** [review]
found an edge case with around 900px width.

image

**Review by keita-determined — COMMENTED** [review]
just found another one. Could you please make sure to test these edge cases before the final review? thanks

image

**Review by keita-determined — COMMENTED** [review]
instead of randomly choosing the breakpoint, can we just build a good HTLM structure to support multiple viewports? The current implementation would cause the similar issues in the future if we add/remove buttons or other components.

File: webui/react/src/hooks/useMobile.ts [hunk 2]
```diff
@@ -8,4 +10,24 @@ const useMobile = (): boolean => {
   return width < MOBILE_BREAKPOINT;
 };
 
+export const useMediaQuery = (mq: string): boolean => {
+  const matchMediaQuery = useSyncExternalStore(subscribe(mq), getSnapshot(mq));
+
+  return matchMediaQuery;
```
Comment (ashtonG) [inline webui/react/src/hooks/useMobile.ts:16]: nit:
```suggestion
  return useSyncExternalStore(subscribe(mq), getSnapshot(mq));
```

File: webui/react/src/pages/Admin/UserManagement.tsx [hunk 3]
```diff
@@ -483,53 +483,59 @@ const UserManagement: React.FC<Props> = ({ onUserCreate }: Props) => {
...
-                  aria-label={CREATE_USER_LABEL}
-                  data-testid="addUser"
-                  disabled={!info.userManagementEnabled || !canModifyUsers}
-                  onClick={CreateUserModal.open}>
-                  {CREATE_USER}
-                </Button>
-              </Row>
-            </Column>
-          </Row>
+          <div className={css.grid}>
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.tsx:486]: can we just make the actionbar the grid?

File: webui/react/src/pages/Admin/UserManagement.tsx [hunk 4]
```diff
@@ -483,53 +483,59 @@ const UserManagement: React.FC<Props> = ({ onUserCreate }: Props) => {
...
+                data-testid="statusSelect"
+                options={statusOptions}
+                placeholder="Status"
+                searchable={false}
+                value={statusFilter}
+                width={'100%'}
+                onChange={handleStatusFilterApply}
+              />
+            </div>
+            <div className={css.gridButton}>
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.tsx:520]: nit: maybe name this `gridButtons` as there may be more than one.

File: webui/react/src/pages/Admin/UserManagement.tsx [hunk 5]
```diff
@@ -483,53 +483,59 @@ const UserManagement: React.FC<Props> = ({ onUserCreate }: Props) => {
...
-                  data-testid="addUser"
-                  disabled={!info.userManagementEnabled || !canModifyUsers}
-                  onClick={CreateUserModal.open}>
-                  {CREATE_USER}
-                </Button>
-              </Row>
-            </Column>
-          </Row>
+          <div className={css.grid}>
+            <div className={css.gridInput}>
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.tsx:487]: are these wrapper divs necessary? my understanding of the grid is that it should automatically place these in the right area.

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 6]
```diff
@@ -11,3 +11,35 @@
...
+.statusSelect {
+  grid-area: statusSelect;
+}
+.gridButton {
+  grid-area: button;
+}
+.grid {
+  align-items: center;
+  column-gap: 8px;
+  display: inline-grid;
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.module.scss:29]:
```suggestion
  display: grid;
```

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 7]
```diff
@@ -11,3 +11,35 @@
...
+  grid-template: 'input roleSelect statusSelect button' auto / minmax(auto, max-content) minmax(auto, 120px) minmax(auto, 170px) auto;
+  width: 100%;
+}
+
+@media only screen and (max-width: $breakpoint-tablet) {
+  .grid {
+    display: grid;
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.module.scss:36]:
```suggestion
```

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 8]
```diff
@@ -11,3 +11,35 @@
...
+}
+.gridButton {
+  grid-area: button;
+}
+.grid {
+  align-items: center;
+  column-gap: 8px;
+  display: inline-grid;
+  grid-template: 'input roleSelect statusSelect button' auto / minmax(auto, max-content) minmax(auto, 120px) minmax(auto, 170px) auto;
+  width: 100%;
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.module.scss:31]:
```suggestion
```

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 9]
```diff
@@ -11,3 +11,35 @@
...
+}
+.statusSelect {
+  grid-area: statusSelect;
+}
+.gridButton {
+  grid-area: button;
+}
+.grid {
+  align-items: center;
+  column-gap: 8px;
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.module.scss:28]:
```suggestion
  gap: var(--spacing-md);
```

File: webui/react/src/hooks/useMobile.ts [hunk 10]
```diff
@@ -8,4 +10,24 @@ const useMobile = (): boolean => {
   return width < MOBILE_BREAKPOINT;
 };
 
+export const useMediaQuery = (mq: string): boolean => {
```
Comment (ashtonG) [inline webui/react/src/hooks/useMobile.ts:13]: given that this isn't in use in the pr, is it necessary to do here?

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 11]
```diff
@@ -11,3 +11,35 @@
...
+@media only screen and (max-width: $breakpoint-tablet) {
+  .grid {
+    display: grid;
+    grid-template: 
+      'input' auto
+      'roleSelect' auto
+      'statusSelect' auto
+      'button' auto
+      / 100%;
+    row-gap: 8px;
```
Comment (ashtonG) [inline webui/react/src/pages/Admin/UserManagement.module.scss:43]:
```suggestion
```

File: webui/react/src/pages/Admin/UserManagement.tsx (hunk 5 above)
Comment (ashtonG) [reply webui/react/src/pages/Admin/UserManagement.tsx]: i tried it and it seems to be fine. can you confirm?

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 12]
```diff
@@ -1,13 +1,42 @@
+.actionBar {
+  align-items: center;
+  display: grid;
+  gap: var(--spacing-md);
+  grid-template:
+    'input roleSelect statusSelect button' auto
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/UserManagement.module.scss:6]:
```suggestion
    'input roleSelect statusSelect buttonGroup' auto
```

File: webui/react/src/pages/Admin/UserManagement.module.scss [hunk 13]
```diff
@@ -1,13 +1,42 @@
...
+}
+.gridButtons {
+  grid-area: button;
+}
+.gridInput {
+  grid-area: input;
+}
+.rightAligned {
+  text-align: right;
+}
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/UserManagement.module.scss:18]: i dont think its used

**Review by keita-determined — COMMENTED** [review]
mostly looks good to me.
only concern is that only 4 letters are visible in the search field and texts in the select boxes are truncated in a certain viewport. it wouldn't be visible if we add one more button. 

image

**Comment by keita-determined** [comment]
> @keita-determined wouldn't a better solve for this be to have the menu switch to mobile mode at tablet sizes? looking at the rest of the table, is the content readable?

Your suggestion is to use mobile view in the tablet size viewport,  right? that sounds good to me

**Review by ashtonG — APPROVED** [review]
